contracts/staking/TokenStaking.sol

```diff
@@ -369,18 +361,64 @@ contract TokenStaking is Initializable, IStaking, Checkpoints {
             application
         );
         require(availableTValue >= amount, "Not enough stake to authorize");
-        authorization.authorized += amount;
-        emit AuthorizationIncreased(
-            stakingProvider,
-            application,
-            fromAmount,
-            authorization.authorized
-        );
-        IApplication(application).authorizationIncreased(
-            stakingProvider,
-            fromAmount,
-            authorization.authorized
+        increaseAuthorizationInternal(stakingProvider, application, amount);
+    }
+
+    /// @notice Increases the authorization of the given staking provider for
+    ///         all applications by all stake amount. Can only be called by
+    ///         the given staking provider’s authorizer.
+    /// @dev Calls `authorizationIncreased` callback on each application to
+    ///      notify the applications about authorization change.
+    ///      See `IApplication`.
+    function increaseAuthorization(address stakingProvider)
+        external
+        override
+        onlyAuthorizerOf(stakingProvider)
+    {
+        StakingProviderInfo storage stakingProviderStruct = stakingProviders[
+            stakingProvider
+        ];
+        bool increased = false;
+        uint96 tStake = stakingProviderStruct.tStake;
+        uint256 authorizedApplications = stakingProviderStruct
+            .authorizedApplications
+            .length;
+
+        for (uint256 i = 0; i < applications.length; i++) {
+            address application = applications[i];
+            if (
+                applicationInfo[application].status ==
+                ApplicationStatus.DISABLED
+            ) {
+                continue;
+            }
+
+            uint96 amount = getAvailableToAuthorize(
+                stakingProvider,
+                application
+            );
+
+            // new application
+            if (amount == tStake) {
+                stakingProviderStruct.authorizedApplications.push(application);
+                authorizedApplications++;
+            }
```

Comment:
Does it make sense to put this if block inside the `if (amount > 0)` block below? This can remove the need to make this check when amount == 0, which in any case wouldn't imply pushing a new app.

Reply:
I'm not fan of nested if(s) and for(s) so basically here I'm trying to make it easier to read in expense of one additional check. If you think it's not great approach then I can change it

Reply:
It's not a strong opinion. If you prefer it this way, I'm fine with that 👍